src/core/mainloop/mainloop.c

```diff
@@ -2003,6 +2005,55 @@ add_entropy_callback(time_t now, const or_options_t *options)
   return ENTROPY_INTERVAL;
 }
 
+/** Periodic callback: if there has been no network usage in a while,
+ * enter a dormant state. */
+static int
+check_network_participation_callback(time_t now, const or_options_t *options)
+{
+  /* If we're a server, we can't become dormant. */
+  if (server_mode(options)) {
+    note_user_activity(now);
+    goto end;
+  }
+
+  /* If we're running an onion service, we can't become dormant. */
+  /* XXXX this would be nice to change, so that we can be dormant with a
+   * service. */
+  if (hs_service_get_num_services() || rend_num_services()) {
+    note_user_activity(now);
+    goto end;
+  }
+
+  /* XXXX Add an option to never become dormant. */
+
+  /* If we have any currently open entry streams other than "linked"
+   * connections used for directory requests, those count as user activity.
+   */
+  /* XXXX make this configurable? */
+  if (connection_get_by_type_nonlinked(CONN_TYPE_AP) != NULL) {
+    note_user_activity(now);
+    goto end;
+  }
+
+  /** Become dormant if there has been no user activity in this long. */
+  /* XXXX make this configurable! */
+#define BECOME_DORMANT_AFTER_INACTIVITY (24*60*60)
```

Comment:
This is rather large value! I was thinking more in the realms of 4-6h. For instance, my tor on my desktop will *never* go dormant event if I never use it because it simply never stays powered on for 24h...

Reply:
So, one factor I had in mind here is that we don't reach the break-even point for consensus diffs (where it would make more sense not to keep the consensus up to date) until well after 24 hours.   I *think* that if you become active only once a day, you might as well be active all the time.  On the other hand, for mobile stuff, the number should be lower.